sdk/core/Azure.Core/src/ClientOptions.cs

```diff
@@ -13,15 +13,66 @@ namespace Azure.Core
     /// </summary>
     public abstract class ClientOptions
     {
-        private HttpPipelineTransport _transport = GetDefaultTransport();
+        private HttpPipelineTransport _transport;
+
+        /// <summary>
+        /// Gets the default set of <see cref="ClientOptions"/>. Changes to the <see cref="Default"/> options would be reflected
+        /// in new instances of <see cref="ClientOptions"/> type created after changes to <see cref="Default"/> were made.
+        /// </summary>
+        public static ClientOptions Default { get; private set; } = new DefaultClientOptions();
+
+        // For testing
+        internal static void ResetDefaultOptions()
+        {
+            Default = new DefaultClientOptions();
+        }
 
         /// <summary>
         /// Creates a new instance of <see cref="ClientOptions"/>.
         /// </summary>
-        protected ClientOptions()
+        protected ClientOptions(): this(true)
         {
-            Retry = new RetryOptions();
-            Diagnostics = new DiagnosticsOptions();
+        }
+
+        internal ClientOptions(bool useDefaults)
+        {
+            if (useDefaults)
+            {
+                Retry = new RetryOptions
+                {
+                    MaxRetries = Default.Retry.MaxRetries,
+                    Delay = Default.Retry.Delay,
+                    MaxDelay = Default.Retry.MaxDelay,
+                    Mode = Default.Retry.Mode,
+                    NetworkTimeout = Default.Retry.NetworkTimeout
+                };
```

Comment:
Nit - I wonder if we should make copy .ctors on those types so we can say `new RetryOptions(Default.Retry)`.  I always feel like it's harder to forget to clone a property with a copy .ctor that's right next to all the property declarations.

Reply:
Great idea!